src/content/docs/en/guides/cms/apostrophecms.mdx

```diff
@@ -6,9 +6,505 @@ stub: true
+  '@apostrophecms/image': ImageWidget
+};
+
+export default widgetComponents;
+```
+
+### Displaying the blog pieces
+To display a page with all of the blog pieces navigate to the `src/templates` folder and create a `BlogIndex.astro` file with the following content:
+
+```js title="src/templates/BlogIndex.astro"
+---
+import dayjs from 'dayjs';
+
+const {
+  page,
+  pieces
+} = Astro.props.aposData;
+---
+
+<section class="bp-content">
+  <h1>{ page.title }</h1>
+
+  <h2>Blog Posts</h2>
+
+  {pieces.map(piece => (
+    <h4>
+      Released On { dayjs(piece.publicationDate).format('MMMM D, YYYY') }
+    </h4>
+    <h3>
+      <a href={ piece._url }>{ piece.title }</a>
+    </h3>
+    <h4>{ piece.authorName }</h4>
+  ))}
+</section>
+```
+
+This file extracts both the page and pieces data from the `aposData` prop. Each piece is linked to their own individual page, which will be created next.
+
+Create a `BlogShow.astro` file in the `src/templates` folder with the following content:
+
+```js title="src/templates/BlogShow.astro"
+---
+import AposArea from '@apostrophecms/apostrophe-astro/components/AposArea.astro';
+import dayjs from 'dayjs';
+
+const { page, piece } = Astro.props.aposData;
+const { main } = piece;
+---
+
+<section class="bp-content">
+  <h1>{ piece.title }</h1>
+  <h3>Created by: { piece.authorName }
+  <h4>
+    Released On { dayjs(piece.publicationDate).format('MMMM D, YYYY') }
+  </h4>
+  <AposArea area={content} />
+</section>
+```
+
+Finally, these templates must be mapped to the corresponding ApostropheCMS page-types. Open the `src.templates/index.js` file and modify it to contain the following:
+
+```js title="src/templates/index.js"
+import HomePage from './HomePage.astro';
+import BlogIndexPage from './BlogIndexPage.astro';
+import BlogShowPage from './BlogShowPage.astro';
+
+const templateComponents = {
+  '@apostrophecms/home-page': HomePage,
+  '@apostrophecms/blog-page:index': BlogIndexPage,
+  '@apostrophecms/blog-page:show': BlogShowPage
+};
+
+export default templateComponents;
+```
+
+### Creating content
+
+If it doesn't exist, create a `.env` file at the root of your Astro project and set the value of the `APOS_EXTERNAL_FRONT_KEY` to the same value you used when bringing your ApostropheCMS instance up. Then, bring the Astro site up using `pnpm run dev`. Login to the site using the credentials that were added during the creation of the ApostropheCMS project. Creating and displaying blog articles is a two step-process:
+
+1. Click on the `Blogs` button in the admin bar to open the blog piece creation modal. Create a new blog piece by clicking on the `New Blog` button in the upper right. Create as many blog articles as desired and then exit the modal.
+2. Click on the `Pages` button in the admin bar. From the page tree modal click on the `New Page` button. In the `Type` dropdown in the right column select `Blog`. Add a title for the page and then click `Publish and View`.
```

Comment:
I **love** the "If it doesn't exist, create..." pattern. Super writing there! I think in this case we can expect that this file exists and the variable is set up because we already asked them to do it at the top of the page.

Also, since the heading is Creating Content, I think we can set the stage with an intro line *about* creating content. For example, we could use the last line of your paragraph but describe generally what is going to happen here, not describe actual button pushes. (I also tried to figure out what would happen, but you'll have to make that sentence accurate. It *sounds* like you write posts individually, and then the new page thing publishes an index of page of these posts? So, an individual blog post isn't published until you (re)publish a page? I'm not sure what writing one more new post after you already have some would look like. I save a piece and then republish that same page again?

Is it possible that this is only really a two-step project when you *initially* set this up? And you don't have to republish a new page again if you add a new post? If so, then I wouldn't describe this as a "two step process" but rather talk individually about what adding a post is like, and publishing an index page (do you do this once and it keeps up?) is like. It would probably be helpful here if it addressed not just what happens to *create* this blog, but when you come back to add a post to it. (See my suggestion below for a possible structure of things I think would be helpful to have here. I'm honestly not sure what Step 2 is meant to do, so I just tried to think of tasks someone with a blog would need to do. Please correct me and my lack of understanding, and maybe thinking task-based vs step-based would help with the thinking here.)

Then, I'm unclear about where I'm logging in, and with what credentials? I don't remember a step where credentials were added. I see we start the astro dev server but then... I don't know where I'm supposed to go. 



```suggestion
### Creating blog posts

Some intro sentence about writing/creating/editing content.

With the Astro dev server running, navigate to  xxxxxx in your browser preview and log in to your dashboard/editor/admin panel.

To add your first post to your blog...

You can repeat this step and add as many posts now as you want. You will also do this every time you want to add a new post.

To publish your posts (does this happen automatically or not? Is this what step 2 is for?)

To edit an existing post....

```